File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.cc
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
	int arr[9] = { 0 };
	int total[7] = { 0 };
	int sum = 0;
	int tmp_sum = sum;
	int tmp1; int tmp2;

	for (int i = 0; i < 9; i++) {
		cin >> arr[i];
		sum += arr[i];
	}

	sort(arr, arr + 9);
	
	tmp_sum = sum;
	for (int i = 0; i < 8; i++) {
		for (int j = i + 1; j < 9; j++) {
			tmp1 = i; tmp2 = j;
			tmp_sum -= arr[tmp1];
			tmp_sum -= arr[tmp2];
				
			if (tmp_sum == 100) {
				for (int i = 0; i < 9; i++) {
					if (i != tmp1 && i != tmp2)
						cout << arr[i] << "\n";
				}
				return 0;

				
				
				
			}
			tmp_sum = sum;
		}
	}
}

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.js
const fs = require("fs");
const input = fs
    .readFileSync(process.platform == "linux" ? "/dev/stdin" : "input.txt")
    .toString()
    .trim()
    .split("\n")
    .map(Number);

const total = input.reduce((acc, num) => acc + num, 0);
let a, b, r1, r2;
let found = false;
for (let i = 0; i < 9; i++) {
    if (found) break;
    for (let j = i + 1; j < 9; j++) {
        if (input[i] + input[j] == total - 100) {
            a = input[i];
            b = input[j];
            r1 = i;
            r2 = j;
            found = true;
            break;
        }
    }
}

const arr = input.filter((_, i) => i !== r1 && i !== r2);
console.log(arr.sort((a, b) => a - b).join("\n"));
